app/javascript/alchemy_admin/spinner.js

```diff
@@ -0,0 +1,37 @@
+import { createHtmlElement } from "alchemy_admin/utils/dom_helpers"
+
+export default class Spinner {
+  constructor(size, color = "currentColor") {
+    this.size = size
+    this.color = color
+    this.spinner = undefined
+  }
+
+  /**
+   * @returns {HTMLElement|undefined}
+   */
+  get el() {
+    return this.spinner
+  }
+  /**
+   *
+   * @param {HTMLElement|undefined} parent
+   */
+  spin(parent) {
+    if (typeof parent === "undefined") {
+      parent = document.body
+    }
+    this.spinner = createHtmlElement(
+      `<alchemy-spinner size="${this.size}" color="${this.color}"></alchemy-spinner>`
+    )
+    parent.append(this.spinner)
```

Comment:
This implies that all parents are jQuery objects. Can we pass DOM Elements into here and change the calling code to pass an DOM Element instead of a jQuery object here?

Reply:
TIL https://developer.mozilla.org/en-US/docs/Web/API/Element/append